src/app/teacher/manage-tags-dialog/manage-tags-dialog.component.ts

```diff
@@ -70,4 +81,28 @@ export class ManageTagsDialogComponent implements OnInit {
       panelClass: 'dialog-md'
     });
   }
+
+  protected delete(tag: Tag): void {
+    this.teacherService.getRuns().subscribe((runs) => {
+      if (confirm(this.getDeleteMessage(runs, tag))) {
+        this.projectTagService.deleteTag(tag);
+      }
+    });
+  }
+
+  private getDeleteMessage(runs: Run[], tag: Tag): string {
+    const numberOfProjectsWithTag = this.getNumberOfProjectsWithTag(runs, tag);
```

Comment:
Shorten variable and function names? Same for ```numProjectsMessage```
```suggestion
    const numProjectsWithTag = this.getNumProjectsWithTag(runs, tag);
```